python/paddle/v2/framework/op.py

```diff
@@ -216,38 +155,57 @@ def __impl__(*args, **kwargs):
         opdesc = method(*args, **kwargs)
         return core.Operator.create(opdesc.SerializeToString())
 
-    __impl__.__doc__ = get_docstring_from_op_proto(op_proto)
-    __impl__.all_input_args = [var.name for var in op_proto.inputs]
-    __impl__.all_output_args = [var.name for var in op_proto.outputs]
-    __impl__.all_attr_args = [attr.name for attr in op_proto.attrs]
-    __impl__.all_not_temp_output_args = [
-        var.name for var in op_proto.outputs if not var.temporary
-    ]
+    return {
+        'method': __impl__,
+        'name': op_proto.type,
+        'all_inputs': [var.name for var in op_proto.inputs],
+        'all_outputs': [var.name for var in op_proto.outputs],
+        'all_attrs': [attr.name for attr in op_proto.attrs],
+        'all_no_temp_outputs':
+        [var.name for var in op_proto.outputs if not var.temporary]
+    }
 
-    return __impl__
 
+class OperatorFactory(object):
+    def __init__(self):
+        self.op_methods = dict()
+        for op_proto in get_all_op_protos():
```

Comment:
if python cache the result of get_all_op_protos, will it be a little faster to create an Operator?